conslogging/conslogging_test.go

```diff
@@ -0,0 +1,80 @@
+package conslogging
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_prettyPrefix(t *testing.T) {
+	testCases := []struct {
+		name          string
+		prefixPadding int
+		prefix        string
+		expected      string
+	}{
+		{
+			name:          "does not truncate if prefixPadding is NoPadding",
+			prefixPadding: NoPadding,
+			prefix:        "github.com/earthly/earthly:80524f0d82a353b3444e83f056207e15f4d5447c+hello-world",
+			expected:      "github.com/earthly/earthly:80524f0d82a353b3444e83f056207e15f4d5447c+hello-world",
+		},
+		{
+			name:          "shortens git SHA if prefix is present",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly:80524f0d82a353b3444e83f056207e15f4d5447c+hello-world",
+			expected:      "g/e/earthly:80524f0+hello-world",
+		},
+		{
+			name:          "keeps branch name",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly:some-feature-branch+hello-world",
+			expected:      "g/e/earthly:some-feature-branch+hello-world",
+		},
+		{
+			name:          "keeps branch name closely resembling sha",
+			prefixPadding: DefaultPadding,
+			prefix:        "/e/hello-world:feedfacecafe",
+			expected:      "/e/hello-world:feedfacecafe",
+		},
+		{
+			name:          "keeps branch name containing special characters /-_",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly:-_/ryan_-/branch-names/-_in-here+hello-world",
+			expected:      "g/e/earthly:-_/ryan_-/branch-names/-_in-here+hello-world",
+		},
+		{
+			name:          "simple target with no path or github info",
+			prefixPadding: DefaultPadding,
+			prefix:        "+run",
+			expected:      strings.Repeat(" ", DefaultPadding-4) + "+run",
+		},
+		{
+			name:          "simple target with path",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly+run",
+			expected:      "github.com/earthly/earthly+run",
+		},
+		{
+			name:          "does not add padding if prefix longer than prefixPadding",
+			prefixPadding: 3,
+			prefix:        "+run",
+			expected:      "+run",
+		},
+		{
+			name:          "negative padding results in no change",
+			prefixPadding: -10,
+			prefix:        "+run",
+			expected:      "+run",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
One extra catch here: `tc` is reused in each loop, which means each test run will use the same memory for that variable. In serial, you usually don't notice (unless you're storing values on `tc` during the test), but it's technically test pollution.

With `t.Parallel()`, the test pollution is much more obvious - only one test case will actually run (and it will run many times), because by the time the first goroutine kicks off, the `tc` variable has already been set to the very last test case.

The fix is to add `tc := tc` between lines 73 and 74 - this allocates new memory for each test so that they're not sharing memory. I should have made that more clear in my original review comments.

Reply:
ah 🤦 I changed this too fast without considering that.
Thanks for pointing it out.

Reply:
No problem, I caught it because I've made that mistake more times than I can count :sweat_smile: 

I should add a helper to `onpar` to fix that, since `onpar`'s whole thing is forcing tests to run in parallel...